### PR TITLE
Add 366 release notes

### PR DESCRIPTION
**Pull request by martint**
Closes https://github.com/trinodb/trino/issues/10181

### CODE REVIEW

File: docs/src/main/sphinx/release/release-366.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Release 366 (XX Dec 2021)
+
+## General
+         
+* Add support for query retries. This feature can be turned on by setting the `retry-policy` 
+  config property or the `retry_policy` session property to `retry`. ({issue}`9361`)
+* Reduce queries lock contention. ({issue}`10246`, {issue}`10239`)
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:7]: how about 

```
* Reduce lock contention during query execution. ({issue}`10246`, {issue}`10239`)
```

File: docs/src/main/sphinx/release/release-366.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+* Query performance improvements. ({issue}`10225`)
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:8]: do we need to include this in RL? may be add some more context? If it's hard to condense in a couple of sentences, how about 

```
* Improve query performance through optimizations for in-memory representation. ({issue}`10225`)
```

File: docs/src/main/sphinx/release/release-366.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+* Reduce query latency for contested clusters or complex queries with multiple stages. ({issue}`10249`)
+* Add CREATE privilege kind to SQL grammar. Note, this permission is not used by any
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:10]: `CREATE`

Note, --> Note that?

File: docs/src/main/sphinx/release/release-366.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+## General
+         
+* Add support for query retries. This feature can be turned on by setting the `retry-policy` 
+  config property or the `retry_policy` session property to `retry`. ({issue}`9361`)
+* Reduce queries lock contention. ({issue}`10246`, {issue}`10239`)
+* Query performance improvements. ({issue}`10225`)
+* Reduce query latency for contested clusters or complex queries with multiple stages. ({issue}`10249`)
+* Add CREATE privilege kind to SQL grammar. Note, this permission is not used by any 
+  existing security systems, but is available for plugins. ({issue}`10206`)
+* Add DENY statement. Note, this statement is not supported by any existing security
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:12]: how about
```
Add support for `DENY` statement in the engine.
``` 

Note, --> Note that?

File: docs/src/main/sphinx/release/release-366.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+* Add support for redirects from Hive to Iceberg. This can be configured with `hive.iceberg-catalog-name`
+  catalog configuration property. ({issue}`10173`)
+
+## Iceberg connector
+
+* Fix incorrect query results for tables partitioned on `binary`. ({issue}`9755`)
+
+## MemSQL connector
+
+* Fix incorrect date result when the value is older than or equals to `1582-10-14`. ({issue}`10054`)
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:31]: date -> `DATE`? cc @ebyhr

File: docs/src/main/sphinx/release/release-366.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  supported for backward compatibility. ({issue}`8383`)
+
+## Hive connector
+
+* Add support for redirects from Hive to Iceberg. This can be configured with `hive.iceberg-catalog-name`
+  catalog configuration property. ({issue}`10173`)
+
+## Iceberg connector
+
+* Fix incorrect query results for tables partitioned on `binary`. ({issue}`9755`)
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:27]:
```
partitioned on `binary` typed fields
```
cc @findepi ?

File: docs/src/main/sphinx/release/release-366.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# Release 366 (XX Dec 2021)
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:1]: date reminder

File: docs/src/main/sphinx/release/release-366.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+## General
+         
+* Add support for query retries. This feature can be turned on by setting the `retry-policy`
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-366.md:5]: should we mention that the retries are "automatic"? i.e. without any client side intervention?

File: docs/src/main/sphinx/release/release-366.md (hunk 5 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-366.md]: @phd3 See https://github.com/trinodb/trino/pull/10145#discussion_r761468833

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/release/release-366.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
+# Release 366 (XX Dec 2021)
+
+## General
+         
+* Add support for automatic query retries. This feature can be turned on by setting the `retry-policy`
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-366.md:5]: Should we note this is not safe for DML queries for now?

File: docs/src/main/sphinx/release/release-366.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
+# Release 366 (XX Dec 2021)
+
+## General
+         
+* Add support for automatic query retries. This feature can be turned on by setting the `retry-policy` 
+  config property or the `retry_policy` session property to `retry`. ({issue}`9361`)
+* Reduce lock contention during query execution. ({issue}`10246`, {issue}`10239`)
+* Improve query performance through optimizations to in-memory representations. ({issue}`10225`)
+* Reduce query latency for contested clusters or complex queries with multiple stages. ({issue}`10249`)
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-366.md:9]: contested?

**Review by losipiuk — APPROVED** [review]

File: docs/src/main/sphinx/release/release-366.md (hunk 10 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-366.md:9]: I replaced it with "contended"

File: docs/src/main/sphinx/release/release-366.md (hunk 9 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-366.md:5]: That should go in the documentation.

**Pull request merged**